airbyte-integrations/connectors/destination-bigquery/src/test-integration/java/io/airbyte/integrations/destination/bigquery/BigQueryDestinationAcceptanceTest.java

```diff
@@ -307,19 +308,24 @@ public boolean requiresDateTimeConversionForNormalizedSync() {
   public void convertDateTime(ObjectNode data, Map<String, String> dateTimeFieldNames) {
     var fields = StreamSupport.stream(Spliterators.spliteratorUnknownSize(data.fields(),
         Spliterator.ORDERED), false).toList();
-    data.removeAll();
+    if (dateTimeFieldNames.keySet().isEmpty()) {
+      return;
+    }
     fields.forEach(field -> {
-      var key = field.getKey();
-      if (dateTimeFieldNames.containsKey(key) && DateTimeUtils.isDateTimeValue(field.getValue().asText())) {
-        switch (dateTimeFieldNames.get(key)) {
-          case DATE_TIME -> data.put(key.toLowerCase(), DateTimeUtils.getEpochMicros(field.getValue().asText()));
-          case DATE -> data.put(key.toLowerCase(), DateTimeUtils.convertToDateFormat(field.getValue().asText()));
+      for (String path : dateTimeFieldNames.keySet()) {
+        var key = field.getKey();
+        if (isKeyInPath(path, key) && DateTimeUtils.isDateTimeValue(field.getValue().asText())) {
```

Comment:
could this method (and the other `convertDateTime` implementations) be structured like this?
```java
for (String path : dateTimeFieldNames.keySet()) {
  if (...) {
    setAtPath(data, path, convertedValue)
  }
}
```

My concern is that relying on `isKeyInPath` could cause weird edge cases

Reply:
@edgao I am not sure that my understanding is correct. Do you mean just move logic from `if(...){}` block to the separated method?

Reply:
sorry, my comment wasn't very clear. I meant to say that checking just `isKeyInPath(path, key)` feels risky; I would prefer to iterate over each path that _could_ be updated (i.e. `for (String path : dateTimeFieldNames.keySet())`) and then do that replacement if necessary. Does that help?

(this also saves us some iterations; currently this method is iterating over every dateTimeFieldName _for each_ field in `data`, and this would mean we could avoid iterating over all the fields)

Reply:
Agree with you. Thanks for the suggestion. Done